polaris/benchmark/_base.py

```diff
@@ -310,6 +294,47 @@ def _compute_checksum(dataset, target_cols, input_cols, split, metrics):
         checksum = hash_fn.hexdigest()
         return checksum
 
+    def verify_checksum(self, md5sum: Optional[str] = None):
+        """
+        Recomputes the checksum and verifies whether it matches the stored checksum.
+        """
+        if md5sum is None:
+            md5sum = self._md5sum
+        if md5sum is None:
+            logger.warning(
+                "No checksum to verify against. Specify either the md5sum parameter or "
+                "store the checksum in the benchmark.md5sum attribute. Skipping!"
+            )
+            return
+
+        # Temporarily reset
+        # Calling self.md5sum will recompute the checksum and set it again
+        self._md5sum = None
+        if self.md5sum != md5sum:
+            raise PolarisChecksumError(
+                f"The specified checksum {md5sum} does not match the computed checksum {self.md5sum}"
+            )
+
+    @computed_field
+    @property
+    def md5sum(self) -> Optional[str]:
+        """Lazily compute the checksum once needed."""
```

Comment:
Is it possible that this might be very slow for large datasets? Should we log something to indicate that we're computing (and caching) the checksum to any confusion around why it might take a surprisingly long time to access the `md5sum`?

Reply:
Good idea!